### ISSUE TITLE
Allow the use of builtin bit functions from ES 3.1

### ISSUE THREAD

**Issue by wizgrav** [KhronosGroup/WebGL#3704]
I'm working on a project where the bulk of work is traversing hierarchies of bitmasks in u32s and having the ability to use hardware instructions to find the LSB MSB and count the set bits would help a lot. Could we bring those functions from ES 3.1 (bitCount, findLSB, findMSB) as an extension to WebGL 2?

**Comment by lexaknyazev** [comment]
FWIW, all math ops (as well as a few others) introduced in GLSL ES 3.10 are equally easy to add since they are already supported in the translator:

- Integer Functions
   - `bitfieldExtract` and `bitfieldInsert`
   - `bitfieldReverse`
   - `bitCount`
   - `findLSB` and `findMSB`
   - `uaddCarry` and `usubBorrow`
   - `umulExtended` and `imulExtended`

- Floating-Point Pack and Unpack Functions
   - `[un]packUnorm4x8`
   - `[un]packSnorm4x8`

- Common Functions
  - `mix` with `genIType`/`genUType`/`genBType`
  - `frexp` and `ldexp`

**Comment by wizgrav** [comment]
@lexaknyazev yeah probably best to pack as many functions as reasonably possible in one extension to avoid polluting the registry

**Comment by kenrussell** [comment]
On 2025-02-06 the WebGL working group resolved to move forward with this extension. It will likely be scoped within the `ANGLE_` namespace and not go through the entire Khronos ratification process.

**Comment by kenrussell** [comment]
Let's leave this issue open for tracking purposes until the extension draft has been produced.

**Comment by wizgrav** [comment]
make your own issue
